spec/helpers/project_helper.rb

```diff
@@ -78,6 +83,92 @@ def load_full_project_implementation(client)
 
           [project, blueprint, spec]
         end
+
+        def project_id(client)
+          if @project_id.nil?
+            if $reuse_integration_project
+              begin
+                project_id = File.read(File.join(CACHE_DIR, GoodData::Environment::ConnectionHelper::DEFAULT_DOMAIN, 'project'))
+                project = client.projects(project_id)
+                unless project.title == "DailyUse Project for gooddata-ruby integration tests - DO NOT DELETE"
+                  project.delete
+                  GoodData.logger.error("Cashed project modified")
+                  fail
+                end
+                GoodData.logger.info("Reusing development project ID: #{project_id}")
+                @project_id = project.pid
+                process_id = File.read(File.join(CACHE_DIR, GoodData::Environment::ConnectionHelper::DEFAULT_DOMAIN, 'process'))
+                project.processes('process_id')
+                GoodData.logger.info("Reusing development process ID: #{process_id}")
+                @process_id = process_id
+                schedule_id = File.read(File.join(CACHE_DIR, GoodData::Environment::ConnectionHelper::DEFAULT_DOMAIN, 'schedule'))
+                project.schedules(schedule_id)
+                GoodData.logger.info("Reusing development project ID: #{schedule_id}")
+                @schedule_id = schedule_id
+              rescue Errno::ENOENT, RestClient::NotFound, RestClient::Gone
+                setup_platform_environment(client)
+                FileUtils.mkpath(File.join(CACHE_DIR, GoodData::Environment::ConnectionHelper::DEFAULT_DOMAIN))
+                File.write(File.join(CACHE_DIR, GoodData::Environment::ConnectionHelper::DEFAULT_DOMAIN, 'project'), @project_id)
+                File.write(File.join(CACHE_DIR, GoodData::Environment::ConnectionHelper::DEFAULT_DOMAIN, 'process'), @process_id)
+                File.write(File.join(CACHE_DIR, GoodData::Environment::ConnectionHelper::DEFAULT_DOMAIN, 'schedule'), @schedule_id)
+              end
+            else
+              setup_platform_environment(client)
+            end
+          end
+          @project_id
+        end
+
+        def process_id(client)
+          project_id(client) if @process_id.nil?
+          @process_id
+        end
+
+        def schedule_id(client)
+          project_id(client) if @schedule_id.nil?
+          @schedule_id
+        end
+
+        def setup_platform_environment(client)
+          begin # rubocop:disable RedundantBegin
+            project = client.create_project(
+              title: "DailyUse Project for gooddata-ruby integration tests - DO NOT DELETE",
+              auth_token: ConnectionHelper::SECRETS[:prod_token]
+            )
+            proc = GoodData::Process.deploy(File.expand_path('../../../spec/data/cc', __FILE__),
```

Comment:
warning: Style/ExpandPathArguments: Use `expand_path('../../spec/data/cc', __dir__)` instead of `expand_path('../../../spec/data/cc', __FILE__)`. (rubocop)